### DISCUSSION TITLE
Instructions don't work

### DISCUSSION THREAD

**Discussion by yeaharh** [jasontaylordev/CleanArchitecture#1136] [General]
I followed the github readme instructions and the instructions generate errors. Useless. Typical of almost every Microsoft based technology that's ever been published. The instructions don't work. What an absolute joke.

**Comment by jasontaylordev** [comment]
> I followed the github readme instructions and the instructions generate errors. Useless. Typical of almost every Microsoft based technology that's ever been published. The instructions don't work. What an absolute joke.

Sounds frustrating! Could you provide more details on the errors you encountered and the steps to reproduce? I'd love to fix the instructions.

**Reply by JonathanCruse** [reply]
I think there might be an issue with version 8.0.4 with __getNSwagProjectMetadata

v8.0.4 does not comment out line 6 in Directory.Build.props as version 8.0.3 did:
`<ArtifactsPath>$(MSBuildThisFileDirectory)artifacts</ArtifactsPath>`
`<!-- <ArtifactsPath>$(MSBuildThisFileDirectory)artifacts</ArtifactsPath> -->`
https://github.com/jasontaylordev/DotNet8NswagIssue

Commenting that one out solved the build error for me. 

@yeaharh was that your issue?

**Reply by jasontaylordev** [reply]
Thanks for the suggestion. I've released [`8.0.5`](https://www.nuget.org/packages/Clean.Architecture.Solution.Template/8.0.5) to resolve this issue.

**Comment by yeaharh** [comment]
My suggestion is to add a section to the instructions that includes "pre-requisites" that need to be installed before the project will run.

cmds rundotnet new install Clean.Architecture.Solution.Template::8.0.5

dotnet new ca-sln --client-framework None --output ca-api

cd ca-api/src/Web

dotnet run

EXEC : warning Number: 2,State:0,Class:20 [D:\djag\deploy\clean-architecture\ca-api\src\Web\Web.csproj]
EXEC : warning Number: 2,State:0,Class:20 [D:\djag\deploy\clean-architecture\ca-api\src\Web\Web.csproj]D:\djag\deploy\clean-architecture\ca-api\src\Web\Web.csproj(35,5): warning MSB3073: The command "dotnet "C:\Users\james\.nuget\packages\nswag.msbuild\14.0.3\buildTransitive\../tools/Net80/dotnet-nswag.dll" run config.nswag /variables:Configuration=Debug" exited with code -1.D:\djag\deploy\clean-architecture\ca-api\src\Web\Web.csproj(41,5): error : NSwag command line tool for .NET Core Net80, toolchain v14.0.3.0 (NJsonSchema v11.0.0.0 (Newtonsoft.Json v13.0.0.0));Visit http://NSwag.org for more information.;NSwag bin directory: C:\Users\james\.nuget\packages\nswag.msbuild\14.0.3\tools\Net80;Executing file 'config.nswag' withvariables 'Configuration=Debug'...;Launcher directory: C:\Users\james\.nuget\packages\nswag.msbuild\14.0.3\tools\Net80;fail: ca_api.Infrastructure.Data.ApplicationDbContextInitialiser[0];An error occurred while initialising the database.;Microsoft.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 52 - Unable to locate a Local Database Runtime installation. Verify that SQL Server Express is properly installed and that the Local Database Runtime feature is enabled.);---> System.ComponentModel.Win32Exception (2): The system cannot find the file specified.;at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction);at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose);at Microsoft.Data.SqlClient.TdsParser.Connect(ServerInfo serverInfo, SqlInternalConnectionTds connHandler, Boolean ignoreSniOpenTimeout, Int64 timerExpire, SqlConnectionString connectionOptions, Boolean withFailover);at Microsoft.Data.SqlClient.SqlInternalConnectionTds.AttemptOneLogin(ServerInfo serverInfo, String newPassword, SecureString newSecurePassword, Boolean ignoreSniOpenTimeout, TimeoutTimer timeout, Boolean withFailover);at Microsoft.Data.SqlClient.SqlInternalConnectionTds.LoginNoFailover(ServerInfo serverInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString connectionOptions, SqlCredential credential, TimeoutTimer timeout);at Microsoft.Data.SqlClient.SqlInternalConnectionTds.OpenLoginEnlist(TimeoutTimer timeout, SqlConnectionString connectionOptions, SqlCredential credential, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance);at Microsoft.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, Boolean applyTransientFaultHandling, String accessToken, DbConnectionPool pool);at Microsoft.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, ObjectpoolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions);at Microsoft.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions);at Microsoft.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection);at Microsoft.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptionsuserOptions, DbConnectionInternal oldConnection);at Microsoft.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection);at Microsoft.Data.ProviderBase.DbConnectionPool.WaitForPendingOpen();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenAsync(CancellationToken cancellationToken, Boolean errorsExpected);at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerDatabaseCreator.<>c__DisplayClass20_0.<<ExistsAsync>b__0>d.MoveNext();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerDatabaseCreator.<>c__DisplayClass20_0.<<ExistsAsync>b__0>d.MoveNext();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerDatabaseCreator.<>c__DisplayClass20_0.<<ExistsAsync>b__0>d.MoveNext();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.ExistsAsync(CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.MigrateAsync(String targetMigration, CancellationToken cancellationToken);at ca_api.Infrastructure.Data.ApplicationDbContextInitialiser.InitialiseAsync() in D:\djag\deploy\clean-architecture\ca-api\src\Infrastructure\Data\ApplicationDbContextInitialiser.cs:line 46;ClientConnectionId:00000000-0000-0000-0000-000000000000;Error Number:2,State:0,Class:20;System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.;---> System.AggregateException: One or more errors occurred. (A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and thatSQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 52 - Unable to locate a Local Database Runtime installation. Verify that SQL Server Express is properly installed and that the Local DatabaseRuntime feature is enabled.));---> Microsoft.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 52 - Unable to locate a Local Database Runtime installation. Verify that SQL Server Express is properly installed and that the Local Database Runtime feature is enabled.);---> System.ComponentModel.Win32Exception(2): The system cannot find the file specified.;at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlExceptionexception, Boolean breakConnection, Action`1 wrapCloseInAction);at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose);at Microsoft.Data.SqlClient.TdsParser.Connect(ServerInfo serverInfo, SqlInternalConnectionTds connHandler, Boolean ignoreSniOpenTimeout, Int64 timerExpire, SqlConnectionString connectionOptions, Boolean withFailover);at Microsoft.Data.SqlClient.SqlInternalConnectionTds.AttemptOneLogin(ServerInfo serverInfo, String newPassword, SecureString newSecurePassword, Boolean ignoreSniOpenTimeout, TimeoutTimer timeout, Boolean withFailover);at Microsoft.Data.SqlClient.SqlInternalConnectionTds.LoginNoFailover(ServerInfo serverInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString connectionOptions, SqlCredential credential, TimeoutTimer timeout);at Microsoft.Data.SqlClient.SqlInternalConnectionTds.OpenLoginEnlist(TimeoutTimer timeout, SqlConnectionString connectionOptions, SqlCredential credential, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance);at Microsoft.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, Boolean applyTransientFaultHandling, String accessToken,DbConnectionPool pool);at Microsoft.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions);at Microsoft.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions);at Microsoft.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection);at Microsoft.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection);at Microsoft.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection);at Microsoft.Data.ProviderBase.DbConnectionPool.WaitForPendingOpen();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenInternalAsync(Boolean errorsExpected, CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.OpenAsync(CancellationToken cancellationToken, Boolean errorsExpected);at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerDatabaseCreator.<>c__DisplayClass20_0.<<ExistsAsync>b__0>d.MoveNext();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerDatabaseCreator.<>c__DisplayClass20_0.<<ExistsAsync>b__0>d.MoveNext();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerDatabaseCreator.<>c__DisplayClass20_0.<<ExistsAsync>b__0>d.MoveNext();--- End of stack trace from previous location ---;at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.ExistsAsync(CancellationToken cancellationToken);at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.MigrateAsync(String targetMigration, CancellationToken cancellationToken);at ca_api.Infrastructure.Data.ApplicationDbContextInitialiser.InitialiseAsync() in D:\djag\deploy\clean-architecture\ca-api\src\Infrastructure\Data\ApplicationDbContextInitialiser.cs:line 46;at ca_api.Infrastructure.Data.InitialiserExtensions.InitialiseDatabaseAsync(WebApplication app) in D:\djag\deploy\clean-architecture\ca-api\src\Infrastructure\Data\ApplicationDbContextInitialiser.cs:line 21;at Program.<Main>$(String[] args) in D:\djag\deploy\clean-architecture\ca-api\src\Web\Program.cs:line 17;at Program.<Main>(String[] args);at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor);at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr);ClientConnectionId:00000000-0000-0000-0000-000000000000;Error Number:2,State:0,Class:20;--- End of inner exception stack trace ---;at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions);at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken);at System.Threading.Tasks.Task.Wait();at NSwag.Commands.ServiceProviderResolver.GetServiceProviderWithHostFactoryResolver(Assembly assembly) in /_/src/NSwag.Commands/HostApplication.cs:line 170;at NSwag.Commands.ServiceProviderResolver.GetServiceProvider(Assembly assembly) in /_/src/NSwag.Commands/HostApplication.cs:line 70;at NSwag.Commands.Generation.AspNetCore.AspNetCoreToOpenApiGeneratorCommandEntryPoint.Process(String commandContent, String outputFile, String applicationName) in /_/src/NSwag.Commands/Commands/Generation/AspNetCore/AspNetCoreToOpenApiGeneratorCommandEntryPoint.cs:line 27;at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor);at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr);--- End of inner exception stack trace ---;at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr);at System.Reflection.MethodBaseInvoker.InvokeWithFewArgs(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture);at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters);at NSwag.AspNetCore.Launcher.Program.Main(String[] args) in /_/src/NSwag.AspNetCore.Launcher/Program.cs:line 132;System.InvalidOperationException: Swagger generation failed with non-zero exit code '1'.;at NSwag.Commands.Generation.AspNetCore.AspNetCoreToOpenApiCommand.RunAsync(CommandLineProcessor processor,IConsoleHost host) in /_/src/NSwag.Commands/Commands/Generation/AspNetCore/AspNetCoreToOpenApiCommand.cs:line 195;at NSwag.Commands.NSwagDocumentBase.GenerateSwaggerDocumentAsync() in /_/src/NSwag.Commands/NSwagDocumentBase.cs:line 270;at NSwag.Commands.NSwagDocument.ExecuteAsync() in /_/src/NSwag.Commands/NSwagDocument.cs:line 67;at NSwag.Commands.Document.ExecuteDocumentCommand.ExecuteDocumentAsync(IConsoleHost host, String filePath) in /_/src/NSwag.Commands/Commands/Document/ExecuteDocumentCommand.cs:line 76;at NSwag.Commands.Document.ExecuteDocumentCommand.RunAsync(CommandLineProcessor processor, IConsoleHost host) in /_/src/NSwag.Commands/Commands/Document/ExecuteDocumentCommand.cs:line 33;at NConsole.CommandLineProcessor.ProcessSingleAsync(String[] args, Object input);at NConsole.CommandLineProcessor.ProcessAsync(String[] args, Object input);at NSwag.Commands.NSwagCommandProcessor.ProcessAsync(String[] args) in /_/src/NSwag.Commands/NSwagCommandProcessor.cs:line 62
The build failed. Fix the build errors and run again.



    On Friday, 8 March 2024 at 07:49:27 am AEST, Jason Taylor ***@***.***> wrote:  
 
 


Thanks for the suggestion. I've released 8.0.5 to resolve this issue.

**Reply by JonathanCruse** [reply]
I think the only thing not mentioned in doc is to install .NET SDK / CLI.

The error says your db is unaccessible, habe you checked connection string in src/Web/appsettings.json?
Propably the local [db](https://learn.microsoft.com/en-us/sql/database-engine/configure-windows/sql-server-express-localdb?view=sql-server-ver16) is not running on your mashine. Tho this is not really a prerequisit, you could also connect to a remote db, so that's up to you.

Does that help?

**Reply by jasontaylordev** [reply]
Thanks, @yeaharh; I've updated the readme to list the prerequisites - .NET 8 and Node.js.

Your project fails to run because SQL Server Express might not be installed or the Local Database Runtime feature is not enabled.

Try running this command instead:

```bash
dotnet new ca-sln --client-framework None --output ca-api --use-sqlite
```

With the `--use-sqlite` option, this will then use SQLite for the database; you won't need to install anything.

**Comment by yeaharh** [comment]
Thank you



    On Tuesday, 12 March 2024 at 03:38:13 am AEST, Jason Taylor ***@***.***> wrote:  
 
 


Thanks, @yeaharh; I've updated the readme to list the prerequisites - .NET 8 and Node.js.

Your project fails to run because SQL Server Express might not be installed or the Local Database Runtime feature is not enabled.

Try running this command instead:
dotnet new ca-sln --client-framework None --output ca-api --use-sqlite
With the --use-sqlite option, this will then use SQLite for the database; you won't need to install anything.
